src/irmin-pack/unix/append_only_file_intf.ml

```diff
@@ -0,0 +1,122 @@
+(*
+ * Copyright (c) 2022-2022 Tarides <contact@tarides.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+open Import
+
+module type S = sig
+  module Io : Io.S
+
+  type t
+
+  val create_rw :
+    path:string ->
+    overwrite:bool ->
+    auto_flush_threshold:int ->
+    auto_flush_callback:(unit -> unit) ->
+    (t, [> Io.create_error ]) result
+  (** Create a rw instance of [t] by creating the file. *)
+
+  val open_rw :
+    path:string ->
+    end_offset:int63 ->
+    dead_header_size:int ->
+    auto_flush_threshold:int ->
+    auto_flush_callback:(unit -> unit) ->
+    (t, [> Io.open_error ]) result
+  (** Create a rw instance of [t] by opening an existing file at [path].
+
```

Comment:
There is a sort-of standard for documentation comments of functions: start with a code block where the function is invoked eg 

```[open_rw ~path ~end_offset ~dead_header_size ~auto_flush_threshold ~auto_flush_callback]```

Then continue with the informal description: `creates a rw instance by opening an existing file at [path].` Then you can document the parameters: `[end_offset] is ...; [dead_header_size] is ...`

I think this is a good documentation standard (but I accept there may be times when people want to use some other format!).